sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3548,17 +3548,14 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
   }
 
   def failedToParseExistenceDefaultAsLiteral(fieldName: String, defaultValue: String): Throwable = {
-    new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1344",
-      messageParameters = Map(
-        "fieldName" -> fieldName,
-        "defaultValue" -> defaultValue))
+    SparkException.internalError(s"parse existence default as literal err," +
```

Comment:
cc @dtenedor Is it really an internal error, please, confirm.

Reply:
@dtenedor kindly ping you. Please, have a look at the PR.

Reply:
@MaxGekk @dtenedor In the latest modification, i removed 44  and the reasons are as follows, thank you for reviewing:
1. the only usage of _LEGACY_ERROR_TEMP_1344 is ResolveDefaultColumns#getExistenceDefaultValues when AnalysisException or MatchError matched when call analyze. 
2. AnalysisException thrown from analyze is already formatted(like INVALID_DEFAULT_VALUE.UNRESOLVED_EXPRESSION, INVALID_DEFAULT_VALUE.SUBQUERY_EXPRESSION) but was swallowed by _LEGACY_ERROR_TEMP_1344.
3. MatchError can be considered a coding bug, so it is separated from AnalysisException as an internal error.